### PR TITLE
Remove `SampledValue`

### PR DESCRIPTION
**Pull request by damonbayer**
While a large amount of work went into implementing `SampledValue`, I ultimately think the codebase is better without it. We haven't been able to take advantage of it in any meaningful way, and it adds a lot of complexity and visual noise to the models. I'm open to reimplementing it (or something similar) on a subset of model components in the future, but attaching it every `RandomVariable` seems unwise.

### CODE REVIEW

File: test/test_iid_random_sequence.py [hunk 1]
```diff
@@ -6,13 +6,15 @@
...
-from pyrenew.metaclass import SampledValue
 from pyrenew.process import IIDRandomSequence, StandardNormalSequence
 from pyrenew.randomvariable import (
     DistributionalVariable,
     StaticDistributionalVariable,
 )
 
+distribution = dist.Cauchy(2, 325.0)
+n = 13532
+
```
Comment (dylanhmorris) [inline test/test_iid_random_sequence.py:17]: Should this be here?

File: test/test_observation_poisson.py [hunk 2]
```diff
@@ -19,4 +19,4 @@ def test_poisson_obs():
     with numpyro.handlers.seed(rng_seed=223):
         sim_pois, *_ = pois(mu=rates)
```
Comment (dylanhmorris) [inline test/test_observation_poisson.py:20]: This can and should now be this, yes? 

```suggestion
        sim_pois = pois(mu=rates)
```

**Review by dylanhmorris — COMMENTED** [review]
A few questions, @damonbayer

File: pyrenew/deterministic/nullrv.py [hunk 3]
```diff
@@ -47,11 +46,10 @@ def sample(
...
-        tuple
-            Containing a SampledValue with None.
+        None
         """
 
-        return (SampledValue(None, t_start=self.t_start, t_unit=self.t_unit),)
+        return None
 
 
 class NullProcess(NullVariable):
```
[the remark this answers is not in the document]
Comment (damonbayer) [reply pyrenew/deterministic/nullrv.py]: Addressed in https://github.com/CDCgov/PyRenew/pull/441/commits/e36f061515ac92d385ea835bbf01ed004a6f1690

File: test/test_iid_random_sequence.py (hunk 1 above)
Comment (damonbayer) [reply test/test_iid_random_sequence.py]: addressed in https://github.com/CDCgov/PyRenew/pull/441/commits/72af339c87b87be0757d2b79a66f7933bdbb3424

File: test/test_observation_poisson.py (hunk 2 above)
Comment (damonbayer) [reply test/test_observation_poisson.py]: addressed in https://github.com/CDCgov/PyRenew/pull/441/commits/3206d053b132e445a797f29fb82148bc7fe43674

**Review by dylanhmorris — APPROVED** [review]
LGTM!

**Pull request merged**